src/app/app.component.ts

```diff
@@ -23,3 +27,89 @@ export class AppComponent {
 
   }
 }
+
+@Component({
+  selector: 'app-calendar-header',
+  template: `
+    <mat-form-field>
+      <mat-select placeholder="Calendar Format" [formControl]="form.controls['calendar']">
+        <mat-option *ngFor="let cal of supportedCalendarFormats" [value]="cal">{{cal}}</mat-option>
+      </mat-select>
+    </mat-form-field>
+    <div class="custom-header">
+      <button mat-icon-button (click)="previousClicked('year')">&lt;&lt;</button>
+      <button mat-icon-button (click)="previousClicked('month')">&lt;</button>
+      <span class="custom-header-label">{{periodLabel}}</span>
+      <button mat-icon-button (click)="nextClicked('month')">&gt;</button>
+      <button mat-icon-button (click)="nextClicked('year')">&gt;&gt;</button>
+    </div>
+  `,
+  styleUrls: []
+})
+export class HeaderComponent<D> implements OnInit {
+  constructor(@Host() private _calendar: MatCalendar<JDNConvertibleCalendar>,
+              private _dateAdapter: DateAdapter<JDNConvertibleCalendar>,
+              @Inject(FormBuilder) private fb: FormBuilder) {
+  }
+
+  form: FormGroup;
+
+  supportedCalendarFormats = JDNConvertibleCalendar.supportedCalendars;
+
+  activeFormat;
+
+  ngOnInit() {
+
+    this.activeFormat = 'Gregorian';
+
+    if (this._dateAdapter instanceof JDNConvertibleCalendarDateAdapter) {
+      this.activeFormat = this._dateAdapter.activeCalendarFormat;
+    }
+
+    // build a form for the calendar format selection
+    this.form = this.fb.group({
+      calendar: [this.activeFormat, Validators.required]
+    });
+
+    // update the selected calendar format
+    this.form.valueChanges.subscribe((data) => {
+      this.convertCalendar(data.calendar);
+    });
+
+  }
+
+  convertCalendar(calendar: 'Gregorian' | 'Julian') {
+
+    if (this._dateAdapter instanceof JDNConvertibleCalendarDateAdapter) {
+
+      const convertedDate = this._dateAdapter.convertCalendarFormat(this._calendar.activeDate, calendar);
+
+      this._calendar.activeDate = convertedDate;
+
+      this._calendar._dateSelected(convertedDate);
+
+      this._calendar.ngAfterContentInit();
```

Comment:
@mmalerba 

After the change of the calendar format, the month view is not updated (today's date is still in the old calendar format and also the new calendar date could be in the previous or next month). 

See <https://www.dropbox.com/s/mdkp6b5bv76uwsd/month_view.mov?dl=0>

If the calendar is toggled, it works. Could the event be triggered from here?

Reply:
Is this a possible solution?

`this._calendar._goToDateInView(convertedDate, 'month');`

However, we want to keep the view consistent and not switch from "year" to "month".

Reply:
view can be updated using `_goToDateInView`, see https://github.com/dhlab-basel/JDNConvertibleCalendarDateAdapter/pull/6/commits/b1891b3cb52ba4d88bc1f42ee82c875f9012a5ca

Reply:
I assume that these methods are not meant to be exposed to the outside. Should we add them to the public api?

Reply:
Did you try injecting the `ChangeDetectorRef` and calling `markForCheck()`?

Reply:
I can inject it but today's date does not get updated (the circle stays with the old date). 

However, if I change the month with the arrow controls (`previousClicked`, `nextClicked`), it gets updated.

Reply:
Ah I see, it's because the views only check today's date on init. In that case I think what you're doing should be ok (calling `ngAfterContentInit`), even though it looks kind of weird at a glance. I would just put a comment explaining why you're doing it.

Reply:
Ok, but probably we will need another solution once `JDNConvertibleCalendar` supports additional calendar formats that would require a reinit because the amount of months could be different and the names would be different too (e.g., if you convert from Gregorian to Hebrew calendar).

Btw: I will be away for the next two weeks having some time off :-)
Thanks for all your helpful advice:-)